XenAdmin/XenSearch/Columns.cs

```diff
@@ -94,8 +94,8 @@ protected bool CheckVMTools(IXenObject o, out GridItemBase item)
             {
                 VM.VirtualisationStatus status = vm.virtualisation_status;
                 if (vm.power_state != vm_power_state.Running ||
-                    status == VM.VirtualisationStatus.OPTIMIZED ||
-                    status == VM.VirtualisationStatus.UNKNOWN)
+                    vm.GetVirtualisationStatus.HasFlag(VM.VirtualisationStatus.OPTIMIZED) ||

```

Comment:
This changed from status to GetVirtualisationStatus, which results in calling the function a second time.